AppBuilder/platform/FilterComplex.js

```diff
@@ -307,149 +317,294 @@ module.exports = class FilterComplex extends FilterComplexCore {
    uiInit() {
       let el = $$(this.ids.querybuilder);
       if (el) {
+         // Clear fields
+         while (el.config.fields.length > 0) {
+            el.config.fields.pop();
+         }
          // Set fields
-         el.define("fields", this.fieldsToQB());
-
-         // Set filters
-         // the filters here are an array of Condition Options for each field.
-         el.config.filters.clearAll();
-         (this.filtersToQB() || []).forEach((filter) => {
-            // .filtersToQB() generates an array of filters for each field,
-            // but the filter.id here is the type of condition rule: contains, not_contains, etc...
-            // we need the ID's to be unique, so we will make them unique to each field by adding
-            // {field.id}_{condition.rule} so make the unique id.
-
-            let type = Object.keys(filter.type)[0];
-            // make sure to only update the filter.id 1x
-            if (filter.id.indexOf(type) == -1) {
-               filter.id = type + "_" + filter.id;
-            }
-
-            // now filter.id is unique.
-
-            el.config.filters.add(filter);
+         (this.fieldsToQB() || []).forEach((f) => {
+            el.config.fields.push(f);
          });
       }
    }
 
-   uiCustomValue($selector) {
-      if (
-         !$selector ||
-         !$selector.config ||
-         !$selector.config.value ||
-         !$selector.config.value.key
-      )
-         return;
-
-      let columnName = $selector.config.value.key;
-      let rule = $selector.config.value.rule;
-      // let value = $selector.config.value.value;
-
-      let $valueElem = $selector.queryView({ customEdit: true });
-      if (!$valueElem) return;
-
-      let field = this._Fields.filter((f) => f.columnName == columnName)[0];
-      if (!field) return;
-
-      if (rule == "in_query" || rule == "not_in_query") {
-         this.uiInQueryValue($valueElem, field);
-      } else if (
-         rule == "in_data_collection" ||
-         rule == "not_in_data_collection"
-      ) {
-         this.uiInDataCollectionValue($valueElem, field);
-      } else if (field.key == "list") {
-         this.uiListValue($valueElem, field);
+   // HACK: have to overwrite Webix Query's function to support our custom input requirement.
+   // HooWoo
+   uiQueryCustomValue() {
+      // Could not require on the top of the page. (Webix cound not found error). Yahoo
+      const Query = require("../../js/webix/components/query/query.js");
```

Comment:
Checkout my comment on the `index.js` file and see if that will work instead of this.